### PR TITLE
add macOS support

### PR DESCRIPTION
**Pull request by sodre**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,21 +9,22 @@ source:
...
   patches:
+    # Set CUDA_ARCHITECTURES to OFF
+    - remove-cuda-architectures-for-llava.patch  # [cuda_compiler_version != "None" ]
     # https://github.com/abetlen/llama-cpp-python/commit/e3941d9c674dbd9891dc3ceda390daeb21f05fd1
     - e3941d9.patch 
 
 build:
   skip: True  # [win]
-  skip: True  # [osx]
+  skip: True  # [linux]  # Save CI/CD resources while testing
```
Comment (sodre) [inline recipe/meta.yaml:19]: Remove this skip after the build finishes on macOS
```suggestion
```

**Comment by YYYasin19** [comment]
Does the overlinking error here arise because we're not using llama.cpp as a dependency but repackaging it here again (accidentally)?

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+MACOSX_DEPLOYMENT_TARGET:   # [osx and x86_64]
+- '11.0'                    # [osx and x86_64]
```
[the remark this answers is not in the document]
Comment (sodre) [reply recipe/conda_build_config.yaml:2]: llama.cpp needs 11.0 or higher to work with Metal.

**Comment by sodre** [comment]
> Does the overlinking error here arise because we're not using llama.cpp as a dependency but repackaging it here again (accidentally)?

We are not depending on the external llama.cpp package yet. The build is still happening with the vendored version of llama.cpp. It is better not to add the llama.cpp dependency until *after* we are able to test all the build options int the build matrix.

The overlooking error is because the `libllava.[dylib,so]` is not properly linking to the vendored `lib llama.[dylib,so]`.

I would like to get OSX-64, osx-arm64, and windows fully building before adding LLava back.

**Comment by sodre** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 3]
```diff
@@ -11,19 +11,19 @@ source:
   patches:
     # https://github.com/abetlen/llama-cpp-python/commit/e3941d9c674dbd9891dc3ceda390daeb21f05fd1
     - e3941d9.patch 
+    - osx-64-pick-discrete.patch  # [osx and x86_64]
```
Comment (sodre) [inline recipe/meta.yaml:14]: @YYYasin19, the GPU detection does not work correctly in MacOS Intel.
this is a patch to fix the issue.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -11,19 +11,19 @@ source:
...
   skip: True  # [linux and cuda_compiler_version == "None"]
+  number: 4
   string: cuda{{ cuda_compiler_version | replace('.', '') }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
   string: cpu_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                                 # [cuda_compiler_version == "None"]
   script:
     - export CMAKE_ARGS="${CMAKE_ARGS} -DLLAVA_BUILD=OFF"
-    - export CMAKE_ARGS="${CMAKE_ARGS} -DLLAMA_CUBLAS=ON"                        # [cuda_compiler_version != "None"]
-    - export CMAKE_ARGS="${CMAKE_ARGS} -DDLLAMA_ACCELERATE=ON -DLLAMA_METAL=ON"  # [osx]
+    - export CMAKE_ARGS="${CMAKE_ARGS} -DLLAMA_CUBLAS=ON"                       # [cuda_compiler_version != "None"]
+    - export CMAKE_ARGS="${CMAKE_ARGS} -DLLAMA_ACCELERATE=ON -DLLAMA_METAL=ON"  # [osx]
```
Comment (sodre) [inline recipe/meta.yaml:25]: fix typo

**Review by sodre — COMMENTED** [review]
@YYYasin19, please feel free to ask questions. I tried to leave comments int the PR is something is not clear.

**Pull request merged**